frontend/model/state.js

```diff
@@ -329,6 +335,232 @@ const getters = {
       return profile.displayName || username
     }
   },
+  // used with graphs like those in the dashboard and in the income details modal
+  groupIncomeDistribution (state, getters) {
+    return groupIncomeDistribution({ state, getters, monthstamp: currentMonthstamp() })
+  },
+  // adjusted version of groupIncomeDistribution, used by the payments system
+  groupIncomeAdjustedDistribution (state, getters) {
+    return getters.groupIncomeAdjustedDistributionForMonth(currentMonthstamp())
+  },
+  groupIncomeAdjustedDistributionForMonth (state, getters) {
+    return monthstamp => groupIncomeDistribution({ state, getters, monthstamp, adjusted: true })
+  },
+  ourPayments (state, getters) {
+    const currency = currencies[getters.groupSettings.mincomeCurrency]
+    const ourUsername = getters.ourUsername
+    const cMonthstamp = currentMonthstamp()
+    const pDate = dateFromMonthstamp(cMonthstamp)
+    const dueIn = lastDayOfMonth(pDate)
+    const monthlyPayments = getters.groupMonthlyPayments
+    const allPayments = getters.currentGroupState.payments
+
+    const sent = (() => {
+      const payments = []
+      // QUESTION: Is this only from this month? What about the previous payments from months ago?
+      //           Same question for received.
```

Comment:
Think I answered this question but the answer was removed? Does this still need answering or can the comment be removed?

(Quick answer again: we only keep track of the current month, and the previous month's payments in memory. After this payments are removed from RAM and should be archived. That archiving part hasn't been implemented yet.)

Reply:
Yes, It can be removed. I added to the payments issue the task "Support fetching old payments (+1 month ago)"